addons/website_crm_score/models/sales_team.py

```diff
@@ -0,0 +1,256 @@
+
+    ratio = fields.Float(string='Ratio')
+    score_team_domain = fields.Char('Domain')
+    leads_count = fields.Integer(compute='_count_leads')
+    assigned_leads = fields.Integer(compute='_assigned_leads')
+    unassigned_leads = fields.Integer(compute='_unassigned_leads')
+    capacity = fields.Integer(compute='_capacity')
+    team_user_ids = fields.One2many('team.user', 'team_id', string='Salesman')
+    user_ids = fields.Many2many('res.users', 'team_user', 'team_id', 'user_id', readonly=True)  # Use for display in Kanban
+    min_for_assign = fields.Integer("Minimum score", help="Minimum score to be automatically assign (>=)", default=0, required=True)
+
+    @api.model
+    def direct_assign_leads(self, ids=[]):
+        ctx = dict(self._context, mail_notify_noemail=True)
+        self.with_context(ctx)._assign_leads()
+
+    @api.model
+    def dry_assign_leads(self, ids=[]):
+        self._assign_leads(dry=True)
+
+    @api.model
+    # Note: The dry mode assign only 50 leads per salesteam for speed issues
+    def assign_leads_to_salesteams(self, all_salesteams, dry=False):
+        shuffle(all_salesteams)
+        haslead = True
+
+        while haslead:
+            haslead = False
+            for salesteam in all_salesteams:
+                domain = safe_eval(salesteam['score_team_domain'], evaluation_context)
+                domain.extend([('team_id', '=', False), ('user_id', '=', False)])
+                domain.extend(['|', ('stage_id.on_change', '=', False), '&', ('stage_id.probability', '!=', 0), ('stage_id.probability', '!=', 100)])
+                leads = self.env["crm.lead"].search(domain, limit=50)
+                haslead = haslead or (len(leads) == 50 and not dry)
+
+                if not leads.exists():
+                    return
+
+                if dry:
+                    for lead in leads:
+                        values = {'lead_id': lead.id, 'team_id': salesteam['id']}
+                        self.env['leads.dry.run'].create(values)
+                else:
+                    leads.write({'team_id': salesteam['id']})
+
+                    # Erase fake/false email
+                    spams = map(lambda x: x.id, filter(lambda x: x.email_from and not checkmail(x.email_from), leads))
+                    if spams:
+                        self.env["crm.lead"].browse(spams).write({'email_from': False})
+
+                    # Merge duplicated lead
+                    leads_done = []
+                    for lead in leads:
+                        if lead.id not in leads_done:
+                            leads_duplicated = lead.get_duplicated_leads(False)
+                            if len(leads_duplicated) > 1:
+                                self.env["crm.lead"].browse(leads_duplicated).merge_opportunity(False, False)
+                            leads_done += leads_duplicated
+                        self._cr.commit()
+                self._cr.commit()
```

Comment:
Do by the cron and can take a long time... very long time... and so don't finish in 20 minutes.
merge_opportunity sent email, and other stuffs that cannot be rollbacked